clang/lib/CIR/CodeGen/CIRGenItaniumCXXABI.cpp

```diff
@@ -2700,4 +2722,56 @@ Address CIRGenItaniumCXXABI::initializeArrayCookie(CIRGenFunction &CGF,
   auto OffsetOp = CGF.getBuilder().getSignedInt(Loc, Offset, /*width=*/32);
   auto DataPtr = CGF.getBuilder().createPtrStride(Loc, CastOp, OffsetOp);
   return Address(DataPtr, NewPtr.getType(), NewPtr.getAlignment());
-}
\ No newline at end of file
+}
+
+CharUnits CIRGenARMCXXABI::getArrayCookieSizeImpl(QualType elementType) {
+  // ARM says that the cookie is always:
+  //   struct array_cookie {
+  //     std::size_t element_size; // element_size != 0
+  //     std::size_t element_count;
+  //   };
+  // But the base ABI doesn't give anything an alignment greater than
+  // 8, so we can dismiss this as typical ABI-author blindness to
+  // actual language complexity and round up to the element alignment.
+  return std::max(CharUnits::fromQuantity(2 * CGM.SizeSizeInBytes),
+                  getContext().getTypeAlignInChars(elementType));
+}
+
+Address CIRGenARMCXXABI::initializeArrayCookie(CIRGenFunction &CGF,
```

Comment:
Use camelBack for all variables in this function

Reply:
The existing style in this file is mixed, but it looks like it's mostly using CamelCase. Do you want me to switch to camelBack for this change in anticipation of the rest of the file being brought into line with a future formatting change?

Reply:
When changing existing functions that haven't been converted to camelCase it makes sense to keep using CamelBack if you patching them, but for new ones it's probably good to go ahead and convert (note that in your other PR I didn't make the same comment). Mixed style in the file level scope it's fine for me, eventually a future formatting change should take care of the rest (just trying not to introduce new ones)